src/cpu_shader/euler.rs

```diff
@@ -0,0 +1,277 @@
+// Copyright 2023 the Vello Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT OR Unlicense
+
+//! Utility functions for Euler Spiral based stroke expansion.
+
+// Use the same constants as the f64 version.
+#![allow(clippy::excessive_precision)]
+
+use super::util::Vec2;
+use std::f32::consts::FRAC_PI_4;
+
+#[derive(Debug)]
+pub struct CubicParams {
```

Comment:
I suggest documenting this parameterization for a cubic and briefly explain how it relates to the Euler spiral, in a few words.